content/en/blog/_posts/2022-04-28-cni-version-related-breaking-change.md

```diff
@@ -0,0 +1,142 @@
+  or later) of the CNI loopback plugin.
+  - Upgrade node components (for example, the kubelet) to Kubernetes 1.24
+  - Upgrade to or install the most current version of the container runtime.
+3. Bring the node back into your cluster by restarting your container runtime
+and kubelet. Uncordon the node.
+
+If you performed the upgrade without following these steps and your network is
+broken, see the following [Troubleshooting](#troubleshooting) section.
+
+## Explanation
+
+CNI enables Kubernetes 1.24 networking. The given container runtimes support the
+most current versions of CNI, v1.0.0 and later. For CNI to work, you must use a
+CNI plugin, and its configuration file must specify the CNI version for which
+the plugin was compiled.
+
+If the version of the plugin is not specified, CNI v1.0.0 determines that the
+plugin is v0.1.0, even if this is not the version. As a DevOps engineer or
+administrator, you must review your CNI configuration files to ensure that they
+specify the correct plugin version. If an incorrect version is specified, errors
+might occur.
+
+You must safely drain and cordon a node before upgrading the CNI plugins on the
+node to a current version. Draining is required because the currently running
+pods have a networking setup based on the older CNI plugins. To correctly tear
+down the network on the running pods, you should leave the older version of the
+CNI Plugins in place to perform the cordoning operation correctly.
+
+## An example containerd configuration file
+
+The following example shows a configuration for `containerd` runtime v1.6.x,
+which supports the latest version of CNI, v1.0.x.
+
+Please see the documentation from your plugin and networking provider for
+further instructions on configuring your system.
+
+By default, the `containerd` runtime adds a loopback interface, `lo`, to
+containers via the CNI plugin. The loopback plugin is distributed with the
+`containerd` release binaries that have the `cni` designation. `containerd`
+v1.6.0 and later includes a CNI v1.0.0-compatible loopback plugin as well as
+other default CNI plugins. The configuration for the loopback plugin is done
+internally and is set to CNI v1.0.0. This means the version of the loopback
+plugin must be v1.0.0 or later when this newer version `containerd` is started.
+
+The following bash command generates a default CNI config. Here, the 1.0.0 value
+for the config version is assigned to the `cniVersion` field for use when
+`containerd` invokes the CNI bridge plugin.
+
+```bash
+cat << EOF | tee /etc/cni/net.d/10-containerd-net.conflist
+{
+ "cniVersion": "1.0.0",
+ "name": "containerd-net",
+ "plugins": [
+   {
+     "type": "bridge",
+     "bridge": "cni0",
+     "isGateway": true,
+     "ipMasq": true,
+     "promiscMode": true,
+     "ipam": {
+       "type": "host-local",
+       "ranges": [
+         [{
+           "subnet": "10.88.0.0/16"
+         }],
+         [{
+           "subnet": "2001:4860:4860::/64"
```

Comment:
```suggestion
           "subnet": "2001:db8:4860::/64"
```
(Use a documentation IP address block, not a real one).

Reply:
Updated. Thanks.